tests/regression/src/lib.rs

```diff
@@ -166,34 +292,20 @@ mod tests {
     fn run_valgrind_test(test_name: &str) {
         let exe_path = std::env::args().next().unwrap();
         let commit_hash = git::get_current_commit_hash();
-        let output_file = create_raw_profile_path(test_name, &commit_hash);
+        let raw_profile = RawProfile::new(test_name, &commit_hash);
```

Comment:
I like the direction this is going! I think we could maybe make things a little more consistent across the APIs

- `Profile::new()` will always shell out to a valgrind command
    - that way you don't have to have all of the separate steps for actually executing the command
    - this will entail moving `build_valgrind_command` interior to `RawProfile`
- Invariant: A successfully constructed profile always has a corresponding file written
    - This file can be accessed with `profile.path() -> &str`
    - This way you aren't responsible for writing the file output anymore, it's taken care of as part of the profile construction.

It's probably useful to think about what the "pseudocode" experience of this would look like, and then picking abstractions to attempt to match that. In my head, the main control flow looks a little something like this:
```rust
match RegressionTestMode::from_env() {
    RegressionTestMode::Valgrind => {
        // RawProfile::new() will query git for the current commit 
        // & execute the test under valgrind
        let raw_profile = RawProfile::new(test_name);
        // maybe AnnotatedProfile::new(&raw_profile) is better?
        let annotate_profile = raw_profile.annotate();
    },
    RegressionTestMode::Diff => {
        // search the file system for annotated profiles of test_name
        // return the profiles in commit-sorted order
        let (older_profile, newer_profile) = AnnotatedProfile::query(test_name)
        let diff_profile = DiffProfile::new(older_profile, newer_profile);
        diff_profile.assert_performance();
    },
    ...
    
```